charts/templates/statefulset.yaml

```diff
@@ -31,83 +31,29 @@ spec:
               {{- toYaml .Values.extraVolumeMounts | nindent 12 }}
             {{- end }}
           env:
-            - name: ML_ADMIN_USER
+            - name: MARKLOGIC_ADMIN_USERNAME
               valueFrom:
                 secretKeyRef:
                   name: {{ include "marklogic.fullname" . }}
                   key: username
-            - name: ML_ADMIN_PASS
+            - name: MARKLOGIC_ADMIN_PASSWORD
               valueFrom:
                   secretKeyRef:
                     name: {{ include "marklogic.fullname" . }}
                     key: password
+            - name: MARKLOGIC_INIT
+              value: "true"
+            - name: REPLICA_COUNT
```

Comment:
What does this do?

Reply:
It create 2 more env variables that are accessible by the start-marklogic.sh scripts. MARKLOGIC_INIT is a variable we weren't using in the script that used to be hard coded into the statfulset.yaml file but is needed in the overall start-marklogic.sh script found in https://github.com/marklogic/marklogic-docker/pull/86. It is a marker that Marklogic should be initialized. REPLICA_COUNT is a env variable I added in order to tell what environment we were in and if we need to join clusters. Checking if a REPLICA_COUNT variable exists in the .sh script is a good way to tell if we are in a K8S deployment or not. It is also useful if there are any situations down the road where a specific number of replicas should trigger a different response from one of the startup scripts.

Reply:
Sorry, I meant specifically the "REPLICA_COUNT" parameter. Remember that the env variables passed to the docker script are the public interface to initializing and running MarkLogic in docker. Personally, I am not a fan of this parameter unless it has some specific meaning. I would prefer to have the docker startup script define a parameter that performs a specific function.